File: app/Http/Controllers/Api/V1/CategoryApiController.php
<?php

namespace App\Http\Controllers\Api\V1;

use App\Http\Controllers\Controller;
use App\Services\CategoryService;
use Illuminate\Http\Request;

class CategoryApiController extends Controller
{
    protected $categoryService;

    public function __construct(CategoryService $categoryService)
    {
        $this->categoryService = $categoryService;
    }

    public function index()
    {
        return $this->categoryService->getAllCategories();
    }

    public function store(Request $request)
    {
        return $this->categoryService->createCategory($request->all());
    }

    public function show($id)
    {
        return $this->categoryService->getCategoryById($id);
    }

    public function destroy($id)
    {
        return $this->categoryService->deleteCategory($id);
    }

    public function update(Request $request, $id)
    {
        return $this->categoryService->updateCategory($id, $request->all());
    }
}

File: spring-project/src/main/java/com/example/demo/controller/CategoryApiController.java
package com.example.demo.controller;

import com.example.demo.service.CategoryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/categories")
public class CategoryApiController {

    private final CategoryService categoryService;

    @Autowired
    public CategoryApiController(CategoryService categoryService) {
        this.categoryService = categoryService;
    }

    @GetMapping
    public ResponseEntity<?> index() {
        return ResponseEntity.ok(categoryService.getAllCategories());
    }

    @PostMapping
    public ResponseEntity<?> store(@RequestBody Map<String, Object> requestData) {
        return ResponseEntity.ok(categoryService.createCategory(requestData));
    }

    @GetMapping("/{id}")
    public ResponseEntity<?> show(@PathVariable Long id) {
        return ResponseEntity.ok(categoryService.getCategoryById(id));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<?> destroy(@PathVariable Long id) {
        categoryService.deleteCategory(id);
        return ResponseEntity.ok().build();
    }

    @PutMapping("/{id}")
    public ResponseEntity<?> update(@RequestBody Map<String, Object> requestData, @PathVariable Long id) {
        return ResponseEntity.ok(categoryService.updateCategory(id, requestData));
    }
}
aaa